lib/rules/id-blacklist.js

```diff
@@ -34,7 +34,7 @@ module.exports = {
         // Helpers
         //--------------------------------------------------------------------------
 
-        var blacklist = context.options;
+        var blacklistRegex = new RegExp("^(" + context.options.join("|") + ")$");
```

Comment:
I definitely thing it would be better to allow any regex pattern, but I am not sure how to do that without 1) breaking existing usages or 2) requiring some sort of indicator for patterns that need to be regexes.

An existing config with `id-blacklist: [2, 'foo']` would catch an Identifier named `foo`, but not one named `foobar`.  With the current implementation, if I just made each one a regex, the same config would now warn on both `foo` and `foobar`. Is this acceptable? I assumed not, which is why I stuck to a less-than-ideal solution that will still catch the same identifiers.